### DISCUSSION TITLE
Retrieve Not Published status reason

### DISCUSSION THREAD

**Discussion by AlexAS81** [OP-TED/eForms-SDK#516] [Q&A] [answered]
Is it possible to recover the reason for the non-publishing of a file somehow?

I am referring to the notices that have been Submitted and have gone to the Not published status because they have been manually rejected by OP for lawfulness.

Does the validation-report operation of the web service Publication API return that information?

TED's "old web service" returned a code with the reason for the Not Published status (OT "For more information, please contact the production support team", WL "The language of the notice is wrong"...).

Thanks in advance

**Comment by ferrakaop** [comment, marked as answer]
This is not currently possible. The OP coordination team will manually email the author/buyer with the reason for not publishing.

We are looking at way to pass this information back to eNotices2 and expose it via the API but it will probably take a few more months before it's available.

Compared to the current/eSentool process, the vast majority of eForms rejections (Validation Failed) will come from CVS, which provides the error messages in SVRL.  Not Published should only affect a small number of notices, though we realise the reasons will also be useful for eSenders to get by API.

**Comment by AlexAS81** [comment]
In the preview environment, it is possible to generate notices that go to the NOT PUBLISHED status?
Can you send us an example eform that reaches that state?

Thanks in advance

**Comment by ferrakaop** [comment]
Please see the note about getting notices into Not Published status on Preview: https://docs.ted.europa.eu/home/eforms/preview/#_known_enotices2_api_issues

You can get in touch via the TED helpdesk to coordinate a rejection for your test notices.

These are the rules that are currently applied for the lawfulness warning: https://docs.ted.europa.eu/eforms/latest/reference/business-rules/index.html#_lawfulness - the easiest way is to have a buyer that is not in the list of "lawful" countries (like the UK) and not select the use of EU funds.

**Comment by AlexAS81** [comment]
We've requested a manual rejection a couple of times but we can't get them to do it from TED. They are always going to published.
